src/citrine/gemd_queries/filter.py

```diff
@@ -48,15 +48,15 @@ class AllIntegerFilter(Serializable['AllIntegerFilter'], PropertyFilterType):
 
     Parameters
     ----------
-    lower: str
+    lower: int
         The lower bound on this filter range.
-    upper: str
+    upper: int
         The upper bound on this filter range.
 
     """
 
-    lower = properties.Float('lower')
-    upper = properties.Float('upper')
+    lower = properties.Integer('lower')
+    upper = properties.Integer('upper')
```

Comment:
You know my thoughts on this from our argument two weeks ago: changing types is an interface break. This is a mild one, as I can't think of a standard operation it would break, since Python's standard division always returns a float, and I'm unsure why a user would do something like `isinstance(value, float)`. But it nonetheless gets into the business of assuming which interface breaks the user cares about, which is a Bad Idea.

I've no intention of wasting more energy on such a petty problem, but I also won't be approving this PR as long as it's left in. Seeing as there are other approvers, and you've already got one from Pablo, it won't stand in your way.

Reply:
I have reverted this fix, as it was tangential to the actual goal of the PR.

In this case, this is not a breaking change because the GemdQuery API is fundamentally broken and does not support any filters, so no user workflow could actually depend on this behavior.